lib/elixir/lib/dynamic_supervisor.ex

```diff
@@ -206,6 +206,12 @@ defmodule DynamicSupervisor do
   See `Supervisor` for more information about child specifications.
   """
   @doc since: "1.6.1"
+  @spec child_spec(options) :: %{
+          id: atom() | term(),
+          start: {DynamicSupervisor, :start_link, args :: [options]},
+          type: :supervisor
+        }
```

Comment:
I think we can simplify the return type. We don't really care about the exact values that are returned for ID or start. We only care that it returns something that complies with the Supervisor.child_spec signature.

Reply:
Done